vyper/venom/basicblock.py

```diff
@@ -373,57 +411,41 @@ def get_read_memory_location(self) -> MemoryLocation:
         if opcode == "mstore":
             return EMPTY_MEMORY_ACCESS
         elif opcode == "mload":
-            if isinstance(self.operands[0], IRLiteral):
-                return MemoryLocation(offset=self.operands[0].value, size=32)
-            return FULL_MEMORY_ACCESS
+            return MemoryLocation.from_operands(self.operands[0], 32)
         elif opcode == "mcopy":
             size, src = self.operands[:2]
-            if isinstance(src, IRLiteral) and isinstance(size, IRLiteral):
-                return MemoryLocation(offset=src.value, size=size.value)
-            return FULL_MEMORY_ACCESS
+            return MemoryLocation.from_operands(src, size)
         elif opcode == "calldatacopy":
             return EMPTY_MEMORY_ACCESS
         elif opcode == "dloadbytes":
             return EMPTY_MEMORY_ACCESS
         elif opcode == "dload":
-            return EMPTY_MEMORY_ACCESS
+            return MemoryLocation(offset=0, size=32)
         elif opcode == "invoke":
-            return FULL_MEMORY_ACCESS
+            return MemoryLocation(offset=0, size=None)
         elif opcode in ("call", "delegatecall", "staticcall"):
             size, dst = self.operands[2:4]
-            if isinstance(dst, IRLiteral) and isinstance(size, IRLiteral):
-                return MemoryLocation(offset=dst.value, size=size.value, is_volatile=False)
-            return FULL_MEMORY_ACCESS
+            return MemoryLocation.from_operands(dst, size, is_volatile=False)
         elif opcode == "return":
             size, src = self.operands
-            if isinstance(src, IRLiteral) and isinstance(size, IRLiteral):
-                return MemoryLocation(offset=src.value, size=size.value, is_volatile=False)
-            return FULL_MEMORY_ACCESS
+            return MemoryLocation.from_operands(src, size, is_volatile=False)
         elif opcode == "create":
             size, src = self.operands[:2]
```

Comment:
full deconstruction:
```suggestion
            size, src, _value = self.operands
```